### PR TITLE
SEP-24: add `refund_memo` and `refund_memo_type` support

### PR DESCRIPTION
**Pull request by JakeUrban**
<details>
  <summary>PR Checklist</summary>

### PR Structure

* [x] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [x] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [x] This PR's title starts with name of package that is most changed in the PR, ex.
  `paymentservice.stellar`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [ ] This PR adds tests for the most critical parts of the new functionality or fixes.
</details>

### What

Adds `refund_memo` and `refund_memo_type` support to SEP-24 `POST /transactions/withdraw/interactive` request.

### Why

This was added to SEP-24. ([#1321](https://github.com/stellar/stellar-protocol/pull/1321))

### CODE REVIEW

**Comment by JakeUrban** [comment]
@lijamie98 I noticed an inconsistencies with the SEP while making this change:

`account` is not required for SEP-24 `POST /transactions/withdraw/interactive`, but the service raises an error if its not specified. It should default to the authenticated account.

I noticed `account` is required for SEP-24 `POST /transations/deposit/interactive` though. This is an inconsistency and we should update the SEP so that `account` is optional just like the withdraw endpoint, and then we'll update the Anchor Platform to match it.

File: core/src/main/java/org/stellar/anchor/util/MemoHelper.java [hunk 1]
```diff
@@ -48,14 +48,14 @@ public static MemoType memoType(Memo memo) {
...
-      memoType = "none";
-    }
-
     if (isEmpty(memo)) {
       return null;
     }
 
+    if (isEmpty(memoType)) {
+      throw new SepValidationException("memo_type is required if memo is specified");
+    }
```
Comment (JakeUrban) [inline core/src/main/java/org/stellar/anchor/util/MemoHelper.java:57]: This is a unrelated change, @lijamie98 let me know if you think I should revert. We should probably require memo type to be specified instead of assuming the type is none, because the error message returned will assume memo type was specified.

**Comment by lijamie98** [comment]
> @lijamie98 I noticed an inconsistencies with the SEP while making this change:
> 
> `account` is not required for SEP-24 `POST /transactions/withdraw/interactive`, but the service raises an error if its not specified. It should default to the authenticated account.
> 
> I noticed `account` is required for SEP-24 `POST /transations/deposit/interactive` though. This is an inconsistency and we should update the SEP so that `account` is optional just like the withdraw endpoint, and then we'll update the Anchor Platform to match it.

Hey @JakeUrban, I will submit a PR for the inconsistency.

The SEP does not specify the account in deposit as optional. https://github.com/stellar/stellar-protocol/blob/master/ecosystem/sep-0024.md#request

I think it is a bug in SEP24. Can you confirm?

File: core/src/main/java/org/stellar/anchor/util/MemoHelper.java (hunk 1 above)
Comment (lijamie98) [reply core/src/main/java/org/stellar/anchor/util/MemoHelper.java:57]: I think we can keep your change that throws exception unless there are cases that we must return `none`.

**Review by lijamie98 — APPROVED** [review]
LGTM.

**Comment by reecexlm** [comment]
not seeing any test coverage
